src/compiler/commandLineParser.ts

```diff
@@ -827,6 +827,15 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
         description: Diagnostics.Do_not_transform_or_elide_any_imports_or_exports_not_marked_as_type_only_ensuring_they_are_written_in_the_output_file_s_format_based_on_the_module_setting,
         defaultValueDescription: false,
     },
+    {
+        name: "isolatedDeclarations",
+        type: "boolean",
+        category: Diagnostics.Interop_Constraints,
+        description: Diagnostics.Ensure_that_each_file_can_have_declaration_emit_generated_without_type_information,
+        defaultValueDescription: false,
+        affectsBuildInfo: true,
+        affectsEmit: true,

```

Comment:
Also `affectsSemanticDiagnostics: true`, since this does have errors emitted in the checker. Honestly, `affectsEmit: false` may actually be more true as well, since afaik this does not actually affect our output.

Reply:
I am unsure how these flags work, however the errors are still emitted in the declaration transform, not the checker. Even if they originate in checker functions, they aren't called in regular type checking, only in during declaration emit. If we set  `affectsSemanticDiagnostics` will it cause the declaration transformer to be called in order to trigger the errors?

Reply:
`affectsSemanticDiagnostics` basically proxies for `affectsDeclarationDiagnostics`, which isn't a standalone thing. The builder logic only copies declaration diagnostics if semantic diagnostics get copied (and `skipLibCheck` is unchanged). See `createBuilderProgramState` in `src/compiler/builder.ts` for the relevant logic. So, name aside, `affectsSemanticDiagnostics` is appropriate here, if ever-so-slightly over-invalidating.

Reply:
Honestly, if we're unsure what level of invalidation gets the desired behavior, a `builder` `verifyTsc`-style unittest is probably not unwarranted.

Reply:
Changed to `affectsSemanticDiagnostics: true`. It functions like other diagnostics that come from declaration emit.